src/app/tracing/services/edit-tracing-settings.service.ts

```diff
@@ -34,14 +34,32 @@ export class EditTracingSettingsService {
         }
     }
 
-    getClearOutbreaksPayload(tracingSettings: TracingSettings, options: ClearOutbreaksOptions): SetTracingSettingsPayload {
+    getClearOutbreaksPayload(tracingSettings: TracingSettings, options: FoodChainElementTypeSelection): SetTracingSettingsPayload {
         return this.getMarkElementsAsOutbreakPayload(tracingSettings, {
-            stationIds: options.clearStationOutbreaks ? tracingSettings.stations.filter(s => s.outbreak).map(s => s.id) : undefined,
-            deliveryIds: options.clearDeliveryOutbreaks ? tracingSettings.deliveries.filter(d => d.outbreak).map(d => d.id) : undefined,
+            stationIds: options.stations ? tracingSettings.stations.filter(s => s.outbreak).map(s => s.id) : undefined,
+            deliveryIds: options.deliveries ? tracingSettings.deliveries.filter(d => d.outbreak).map(d => d.id) : undefined,
             outbreak: false
         });
     }
 
+    getClearKillContaminationsPayload(
+        tracingSettings: TracingSettings,
+        options: FoodChainElementTypeSelection
+    ): SetTracingSettingsPayload {
+
+        const updateFun = <T extends ElementTracingSettings>(x: T) => ({ ...x, killContamination: false });
+
+        tracingSettings = { ...tracingSettings };
+        if (options.stations) {
+            tracingSettings.stations = tracingSettings.stations.map(updateFun);
+        }
+        if (options.deliveries) {
+            tracingSettings.deliveries = tracingSettings.deliveries.map(updateFun);
+        }
```

Comment:
Nitpick:
I would prefer renaming this function, the suffix `Fun` is not consistently used for functions in our codebase. 
I would either simply call this function `update `or if that isn't clear enough perhaps `handleUpdate`.